src/cacheout/cache.py

```diff
@@ -513,7 +537,7 @@ def _popitem(self):
             raise KeyError("popitem(): cache is empty")
 
         value = self._cache[key]
-        self._delete(key)
+        self._delete(key, EvictedCause.SIZE)
```

Comment:
I wonder if `_popitem()` might need the eviction cause as an argument since it could potentially be called for different reasons. It's either called from `evict()` which would be SIZE but when called from `popitem()` it could be considered EXPLICIT since it's similar to a direct call to `delete()` except the key isn't being supplied. Seems like SIZE might not fit so well if the caller is the one making the `popitem()` call (so really the cache might not be full, it's just being intentionally pruned based on the cache policy).

Just brainstorming here, but another option might be to have POLICY replace SIZE which would encompass deleting the key because either the cache was full or because the key was removed via `popitem()`. But maybe wanting to differentiate between cache-full and an explicit call to remove a key would be preferable.

Thoughts?